### ISSUE TITLE
Override Python language version (kernel) for all notebooks in a Dev Container

### ISSUE THREAD

**Issue by pamelafox** [microsoft/vscode-jupyter#13641]
I have created a Dev Container for this repo:

https://github.com/pamelafox/materials-fa22

When I open a notebook in that repo, I get prompted to pick a language version. It seems to have both 3.9 and 3.10 in the environment - I think 3.9 comes from the base Debian image, but 3.10 is the desired version for the notebooks. 

<img width="988" alt="Screenshot 2023-06-02 at 11 52 46 AM" src="https://github.com/microsoft/vscode-jupyter/assets/297042/f086a0e4-8593-4fb9-b9de-70c69982180e">

When I do pick 3.10, it saves it into the notebook's metadata as the version, and it then recommends that version going forward for newly opened notebooks.

-   "version": "3.9.13"
+   "version": "3.10.11"
   }

I would love to be able to just specify the version at the dev container level, in a setting for the Jupyter extension, as I don't want students using the repo to accidentally pick 3.9 instead of 3.10. I don't want them to have to make that decision at all, since it can be a difficult decision for someone new to Python. I also don't want to have to re-save every notebook with the exact version every time we update the Dev Container language version.

Thanks!

**Comment by jtkiley** [comment]
I agree that it would be nice to set this in some sort of in-repo configuration that will get pulled in by a devcontainer, perhaps combined with a reasonable default inference.

I suspect that the thing we'd actually want to specify is the path to the executable and not the version. That's because, across multiple computers, the patch version can often be different because devcontainers tend to come only at a major/minor level of specificity (and pulling the then-current patch when built).

Additionally, it's probably a reasonable inference that someone using a devcontainer image that specifies, for example, Python 3.11 will want `/usr/local/bin/python` and not `usr/bin/python3`, so this might be something that could be set at or inferred from the image level, making the right choice for most users without configuration (and being easily configurable when needed).

**Comment by JosephDHenry** [comment]
This is needed by my company as well. It would be nice if it simply defaulted to the default python path.

**Comment by DonJayamanne** [comment]
Duplicate https://github.com/microsoft/vscode/issues/130946
